src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp

```diff
@@ -0,0 +1,467 @@
+    case miopenInt8: Init<int8_t>(problem); break;
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: break;
+    }
+#endif
+}
+
+bool PerformanceConfigHipImplicitGemmGroupFwdXdlops::SetNextValue(const ProblemDescription& problem)
+{
+    if(valid_kernels.empty())
+    {
+        this->HeuristicInit(problem);
+        assert(!valid_kernels.empty());
+        return true;
+    }
+    if((index + 1) < valid_kernels.size())
+    {
+        ++index;
+        this->kernel_id = this->valid_kernels[index];
+        return true;
+    }
+    else
+        return false;
+}
+
+bool PerformanceConfigHipImplicitGemmGroupFwdXdlops::IsValidValue() const
+{
+    return index < valid_kernels.size();
+}
+
+bool PerformanceConfigHipImplicitGemmGroupFwdXdlops::IsValid(
+    const ProblemDescription& problem) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = problem;
+    return false;
+#else
+    switch(problem.conv_problem.GetInDataType())
+    {
+    case miopenHalf: return CheckIsSupportCKArgs<ck::half_t>(problem);
+    case miopenFloat: return CheckIsSupportCKArgs<float>(problem);
+    case miopenInt8: return CheckIsSupportCKArgs<int8_t>(problem);
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: break;
+    }
+    return false;
+#endif
+}
+
+bool PerformanceConfigHipImplicitGemmGroupFwdXdlops::operator==(
+    const PerformanceConfigHipImplicitGemmGroupFwdXdlops& other) const
+{
+    return this->kernel_id == other.kernel_id;
+}
+
+PerformanceConfigHipImplicitGemmGroupFwdXdlops
+ConvHipImplicitGemmGroupFwdXdlops::GetDefaultPerformanceConfig(
+    const ConvolutionContext&, const ProblemDescription& problem) const
+{
+    PerformanceConfigHipImplicitGemmGroupFwdXdlops pp;
+    pp.HeuristicInit(problem);
+    return pp;
+}
+
+bool ConvHipImplicitGemmGroupFwdXdlops::IsValidPerformanceConfig(
+    const ConvolutionContext&,
+    const ProblemDescription& problem,
+    const PerformanceConfigHipImplicitGemmGroupFwdXdlops& config) const
+{
+    return config.IsValid(problem);
+}
+
+PerformanceConfigHipImplicitGemmGroupFwdXdlops
+ConvHipImplicitGemmGroupFwdXdlops::Search(const ConvolutionContext& ctx,
+                                          const ProblemDescription& problem,
+                                          const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, ctx, problem, invoke_ctx);
+}
+
+bool ConvHipImplicitGemmGroupFwdXdlops::IsApplicable(const ConvolutionContext& ctx,
+                                                     const ProblemDescription& problem) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    std::ignore = problem;
+    return false;
+#else
+    if(miopen::IsDisabled(MIOPEN_DEBUG_GROUP_CONV_IMPLICIT_GEMM_HIP_FWD_XDLOPS{}))
+        return false;
+    if(miopen::IsEnabled(MIOPEN_DEBUG_CONVOLUTION_DETERMINISTIC{}))
+        return false;
```

Comment:
[Correctness] This is wrong as it does not obey the attribute set in the convolution descriptor.
```C++
    if(problem.conv_problem.GetConv().attribute.deterministic)
        return false;
```

Reply:
[resolved]